### PR TITLE
expression: make "SHOW TABLE STATUS" be case insensitive

### LINKED ISSUE #7518: SHOW TABLE STATUS case sensitivity
Please answer these questions before submitting your issue. Thanks!

1. What did you do?

```
MySQL [test]> show tables;
Empty set (0.00 sec)

MySQL [test]> CREATE TABLE t1 (a int);
Query OK, 0 rows affected (0.12 sec)

MySQL [test]> SELECT * FROM T1;
Empty set (0.11 sec)

MySQL [test]> SHOW TABLE STATUS like 't1';
+------+--------+---------+------------+------+----------------+-------------+-----------------+--------------+-----------+----------------+-----------------------------------+---------------------+---------------------+-----------------+----------+----------------+---------+
| Name | Engine | Version | Row_format | Rows | Avg_row_length | Data_length | Max_data_length | Index_length | Data_free | Auto_increment | Create_time                       | Update_time         | Check_time          | Collation       | Checksum | Create_options | Comment |
+------+--------+---------+------------+------+----------------+-------------+-----------------+--------------+-----------+----------------+-----------------------------------+---------------------+---------------------+-----------------+----------+----------------+---------+
| t1   | InnoDB |      10 | Compact    |  100 |            100 |         100 |             100 |          100 |       100 |            100 | 2018-08-28 14:31:40.165 +0000 UTC | 2018-08-28 14:32:04 | 2018-08-28 14:32:04 | utf8_general_ci |          |                |         |
+------+--------+---------+------------+------+----------------+-------------+-----------------+--------------+-----------+----------------+-----------------------------------+---------------------+---------------------+-----------------+----------+----------------+---------+
1 row in set (0.00 sec)

MySQL [test]> SHOW TABLE STATUS like 'T1';
Empty set (0.00 sec)
```
Also:
```
MySQL [test]> SELECT * FROM information_schema.tables where table_schema='test' and table_name='T1'\G
Empty set (0.02 sec)
```

This command is however correct:
```
MySQL [test]> SHOW CREATE TABLE T1;
+-------+------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                         |
+-------+------------------------------------------------------------------------------------------------------+
| t1    | CREATE TABLE `t1` (
  `a` int(11) DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin |
+-------+------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

(I checked some other commands too: `SHOW INDEXES FROM T1` is also correct leading to some inconsistency.)

2. What did you expect to see?

`SHOW TABLE STATUS` should be case insensitive / behave the same as the select statement, since TiDB behaves according to `lower_case_table_names=2`.

3. What did you see instead?

No result.

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

MySQL [test]> SELECT tidb_version()\G
*************************** 1. row ***************************
tidb_version(): Release Version: v2.1.0-beta-229-g5404e2e
Git Commit Hash: 5404e2eaae27b9ce31264a4eff0a41aa92ffe328
Git Branch: master
UTC Build Time: 2018-08-14 01:46:09
GoVersion: go version go1.10.2 linux/amd64
Race Enabled: false
TiKV Min Version: 2.1.0-alpha.1-ff3dd160846b7d1aed9079c389fc188f7f5ea13e
1 row in set (0.00 sec)

### PR DESCRIPTION
**Pull request by iamzhoug37**
### What problem does this PR solve? 

fix #7518

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update the `tidb-ansible` repository
 - Need to be included in the release note

### CODE REVIEW

**Comment by winoros** [comment]
In my opinion, we'd better do this out of the `LIKE` builtin function?
PTAL @lysu @jackysp

**Comment by iamzhoug37** [comment]
OK，let me rewrite this change

**Comment by zz-jason** [comment]
@iamzhoug37 please fill the PR description in detail and check the **Check List** carefully.

File: expression/builtin_like.go [hunk 1]
```diff
@@ -14,7 +14,10 @@
 package expression
 
 import (
```
Comment (zz-jason) [inline expression/builtin_like.go]: please reorg the package imported, separate the standard and the third-part packages:
```go
import (
	"regexp"
	"strconv"
	"strings"

	"github.com/pingcap/tidb/sessionctx/variable"
)
```

File: expression/builtin_like.go [hunk 2]
```diff
@@ -80,6 +83,13 @@ func (b *builtinLikeSig) evalInt(row chunk.Row) (int64, bool, error) {
 		return 0, isNull, errors.Trace(err)
 	}
 	escape := byte(val)
+
+	lowerCaseTableName,_ := strconv.Atoi(variable.SysVars["lower_case_table_names"].Value)
+	if lowerCaseTableName != 0 {
```
Comment (zz-jason) [inline expression/builtin_like.go]: how about directly: `variable.SysVars["lower_case_table_names"].Value != "0"`

**Comment by zz-jason** [comment]
@iamzhoug37 
1. Please add some test to cover the bug this PR fixes.
2. Please fix CI.

**Comment by iamzhoug37** [comment]
Ok，I will fix this. This is my first open source code develop experience, don't know some rules. Please  forgive me..

**Comment by iamzhoug37** [comment]
@winoros If don't implement this logic in `LIKE` builtin function, we need to do this in `ShowExec fetchShowTableStatus` and `expressionRewriter Leave` function. Should we implement this in two place or do you have other better suggestion?

File: expression/evaluator_test.go [hunk 3]
```diff
@@ -14,6 +14,7 @@
 package expression
 
 import (
+	"github.com/pingcap/tidb/sessionctx/variable"
```
Comment (zz-jason) [inline expression/evaluator_test.go]: please reorg the packages.

Comment (iamzhoug37) [reply expression/evaluator_test.go]: Ok，I'm doing "make dev" in my local

**Pull request closed without merging**